salvo/src/lib/benchmark/base_benchmark.py

```diff
@@ -105,40 +164,32 @@ def run_image(self, image_name: str,
     """
     return self._docker_image.run_image(image_name, run_parameters)
 
-  def pull_images(self) -> List[str]:
-    """Retrieve all images necessary for the benchmark.
+  @abc.abstractmethod
+  def execute_benchmark(self) -> bool:
```

Comment:
Returning a bool lacks the ability to give a good error message. The workflow will just say "failed to execute benchmark" and the user will have to read the logs to find the actual failure.

Would it be better to change this to returning None and raising an exception instead?

Reply:
Give me a few minutes :)  Let me update things.